LBHD, Directional Armor v1.2.lsl

```diff
@@ -0,0 +1,206 @@
+integer los(vector start, vector target)
+{
+    list ray=llCastRay(target,start,[RC_REJECT_TYPES,RC_REJECT_AGENTS,RC_DATA_FLAGS,RC_GET_ROOT_KEY,RC_MAX_HITS,1]);
+    key hit=llList2Key(ray,0);//Debug
+    if(llKey2Name(hit))
+    {
+        //llSay(0,llKey2Name(hit));//Debug
+        if(hit==me)return 1;
+        else return 0;//Object in way
+    }
+    else
+    {
+        if(llList2Vector(ray,1)==ZERO_VECTOR)return 1;
+        else return 0;//Land in way
+    }
+}
+update()//SetText
+{
+    llSetLinkPrimitiveParamsFast(-4,[PRIM_TEXT,"[LBHD]\n "+(string)hp+" / "+(string)mhp+" HP",<0.0,0.75,1.0>,1.0,
+        PRIM_DESC,"LBA.v.LBHD"+(string)hp+","+(string)mhp+","+(string)atcap+",999"+
+        //In order: Current HP, Max HP, Max AT accepted, Max healing accepted (Not implemented)
+        ",F-"+llGetSubString((string)front,0,2)+//Frontal modifier
+        ",S-"+llGetSubString((string)side,0,2)+//Side modifier
+        ",R-"+llGetSubString((string)back,0,2)+//Rear modifier
+        ",T-"+llGetSubString((string)top,0,2)+//Top modifier
+        ",B-"+llGetSubString((string)bottom,0,2)+//Bottom modifier
+        ",M-"+llGetSubString((string)middle,0,2)]);//Middle Modifier
+        //Top and bottom are present for later use but are currently not implemented.
+}
+die()
+{
+    //Add extra shit here
+    //llResetScript();//Debug
+    llDie();//Otherwise, use this
+}
+vector tar(key id)
+{
+    vector av=(vector)((string)llGetObjectDetails(id,[OBJECT_POS]));
+    return av;
+}
+key user;
+key gen;//Object rezzer
+key me;
+integer hear;
+default
+{
+    state_entry()
+    {
+        user=llGetOwner();
+        me=llGetKey();
+        gen=(string)llGetObjectDetails(me,[OBJECT_REZZER_KEY]);
+        update();
+        //llListen(-500,"","","");//Non-Hex
+        integer hex=(integer)("0x" + llGetSubString(llMD5String((string)me,0), 0, 3));
+        hear=llListen(hex,"","","");
+        llSetTimerEvent(5.0);//Used for auto-delete.
+    }
+    on_rez(integer p)
+    {
+        if(p>1)//Allows HUD/Objects to set HP value when rezzed with a param, otherwise uses default
+        {
+            mhp=p;
+            hp=p;
+        }
+        me=llGetKey();
+        integer hex=(integer)("0x" + llGetSubString(llMD5String((string)me,0), 0, 3));
+        if(hear)llListenRemove(hear);
+        hear=llListen(hex,"","","");
+        gen=(string)llGetObjectDetails(me,[OBJECT_REZZER_KEY]);
+        llSetTimerEvent(5.0);//Used for auto-delete.
+        user=llGetOwner();
+        update();
+    }
+    listen(integer chan, string name, key id, string message)
+    {
+        //[ALWAYS] USE llRegionSayTo(). Do not flood the channel with useless garbage that'll poll every object in listening range.
+        list parse=llParseString2List(message,[","],[" "]);
+        if((key)llList2String(parse,0)==me)//targetcheck
+        {
+            vector pos=llGetPos();
+            vector tpos=tar(id);
+            //if(los(pos,tpos))//Enforces LBA line-of-sight
```

Comment:
This conditional statement is commented out. Its following code block needs to also be commented or entirely removed.

Reply:
That condition is there in case people want to force line of sight check on damage sources. Some people take issue with that idea which is why it's commented out, but can still be re-enabled.